File: app/src/main/java/android/matrix/ooredoo/qa/problems/java/BinaryGap.java
package android.matrix.ooredoo.qa.problems.java;

public class BinaryGap {

    public int solution(int n) {
       String binaryumber =  Integer.toBinaryString(n);

       int gap = 0;
       int zeroCount = 0;

        for (char b:
             binaryumber.toCharArray()) {
            if (b == '1') {
                if (zeroCount > gap) {
                    gap =  zeroCount;
                }
                zeroCount = 0;
            } else {
                zeroCount ++;
            }
        }

        return gap;
    }
}

File: app/src/main/java/android/matrix/ooredoo/qa/problems/kotlin/BinaryGap.kt
package android.matrix.ooredoo.qa.problems.kotlin
class BinaryGap {

    fun solution(n: Int): Int {

        val binaryumber : String = Integer.toBinaryString(n)

        var gap = 0
        var zeroCount = 0

        var b1: String = "apple"


        for (b in binaryumber) {
            if (b == '1') {
                if (zeroCount > gap) {
                    gap = zeroCount
                }
                zeroCount = 0
            } else {
                zeroCount++
            }
        }

        return gap
    }
}
